### ISSUE TITLE
[Feature request] Allow passing tags as a ruby hash

### ISSUE THREAD

**Issue by jtzemp** [DataDog/dogstatsd-ruby#106]
From the README, the suggested way to tag a metric is to pass an array of strings.
```
# Tag a metric.
statsd.histogram('query.time', 10, :tags => ["version:1"])
```
How hard would it be to add the ability to pass a hash to `:tags` in addition to using an Array?

```
# Tag a metric.
statsd.histogram('query.time', 10, :tags => {version: 1})
```

Is this something you'd be willing to add, or if you don't want to add it, would you be willing to consider adding it if I created a PR?

**Comment by jtzemp** [comment]
So, I went ahead and did a PR to add tagging support. See #107.

**Comment by swssqewe12** [comment]
@jtzemp A tag doesn't always have to be a key value pair, it can just be a value. So I can see why they stick with arrays.

**Comment by jtzemp** [comment]
@swssqewe12, very true that tags don't need to be a pair. In the implementation I wrote for the PR I handle that in 2 ways:
* You can pass either an array, or a hash to `tags:`. So nothing needs to change in existing code. If you like arrays, keep using them. If you want a hash, use it. Wanna switch back and forth? Use whatever method best suits the situation.
* If you want to use a hash and send a tag without a value, just set the value in the hash to `nil`.

**Comment by albertvaka** [comment]
Closing as this has been released in 4.3.0
